### PR TITLE
Add used referral code cleanup. Also fix unique index on users.

### PR DESCRIPTION
**Pull request by eugene-eko2000**

### CODE REVIEW

File: be/user.go [hunk 1]
```diff
@@ -63,5 +63,10 @@ func UpdateUser(c *gin.Context) {
 			return
 		}
 	}
+
+	if user.Referral != "" {
+		// TODO: Make the call async after the db connection is handled by the db controller
+		cleanupReferral(db, user.Referral)
```
Comment (eldarm) [inline be/user.go:69]: ? So, a referral code is used only once? That's not our original design.
Also what happens when updating user?

**Review by eldarm — CHANGES_REQUESTED** [review]

File: db/init.sql [hunk 2]
```diff
@@ -14,7 +14,8 @@ CREATE TABLE IF NOT EXISTS users (
   agree_toc varchar(255),
   referral VARCHAR(32),
   ts TIMESTAMP default current_timestamp,
-  PRIMARY KEY (id)
+  PRIMARY KEY (id),
+  UNIQUE INDEX avatar_idx (avatar)
```
[the remark this answers is not in the document]
Comment (eugene-eko2000) [reply db/init.sql:18]: Surlrisingly we lost this change.

File: be/user.go (hunk 1 above)
Comment (eugene-eko2000) [reply be/user.go:69]: Good catch, the index by ref_value shouldn't be unique. I changed that.

And yes, the refcode can be used once by the user i.e. one device at certain IP address. The ref code can be used multiple times by different devices and IP addresses.

File: db/init.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (eldarm) [reply db/init.sql:18]: Ouch... ok...

File: be/user.go (hunk 1 above)
Comment (eldarm) [reply be/user.go:69]: Good.

**Review by eldarm — APPROVED** [review]

**Pull request merged**